felix.go

```diff
@@ -186,6 +186,19 @@ configRetry:
 			time.Sleep(1 * time.Second)
 			continue configRetry
 		}
+
+		// We now have some config flags that affect how we configure the syncer.
+		// After loading the config from the datastore, reconnect, possibly with new
+		// config.  We don't need to re-load the configuration _again_ because the
+		// calculation graph will spot if the config has changed since we were initialised.
+		datastoreConfig = configParams.DatastoreConfig()
```

Comment:
You're correct that the config may have changed between loading it above and this line.  However, we reload the config again as part of the snapshot processing.  At the end of the snapshot, we compare the snapshot-loaded config against the values above.